### ISSUE TITLE
Bloom filters on Int8 and Int16 columns always return false negatives

### ISSUE THREAD

**Issue by progval** [apache/arrow-datafusion#9779]
### Describe the bug

Bloom filters on these columns always filter out every value.

### To Reproduce

#9778 demonstrates this, through `correct_bloom_filters: false` as macro "parameter".

**Comment by progval** [comment]
I just reproduced the bug in the `parquet` crate, so this isn't an issue in Datafusion

**Comment by alamb** [comment]
It turns out that https://github.com/apache/arrow-datafusion/pull/9770 demonstrates that the unsigned variants are incorrect as well so I updated the comments

**Comment by progval** [comment]
Not exactly: as long as #9770 is not merged, bloom filters are not used on `UInt8` and `UInt16`.

Now that I say it, I realize that I should probably amend that PR (and the existing code) to disable bloom filters entirely on these types; so Datafusion is slow instead of incorrect.

**Comment by alamb** [comment]
It seems like https://github.com/apache/arrow-datafusion/pull/9770 was just merged . 

I filed https://github.com/apache/arrow-datafusion/issues/9914 to fix the problem by disabling this feature

**Comment by alamb** [comment]
@edmondop  disabled this code https://github.com/apache/arrow-datafusion/pull/9969 🙏
